app/src/main/java/com/example/rickandmortyapp/ui/character/CharacterFragment.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.rickandmortyapp.ui.character
```

Comment:
This is the Fragment containing the Character logic that was once in the MainActivity before. 

It's 100% tied to the ViewModel to be able to receive data. When the Fragment is created, it will attach itself to the ViewModel viewState flow and reactively update the UI when the ViewState changes.

I've noticed that you declared the binding as a high level property inside the MainActivity. Don't do this. Keeping a binding reference is a memory leak entry point. Inflate the binding, attach it to the data listeners and don't keep it inside the Fragment or Activity. Just like this Fragment does.